### PR TITLE
Ensure role names are compatible with newer ansible

### PR DESCRIPTION
**Pull request by cjeanner**
Ansible doesn't want dash in role names anymore - so we must switch
them to underscore instead[1].

But since those roles are probably used as-is in external location, we
ensure backward compatiblity with existing tools by using symlinks.

[1] https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

### CODE REVIEW

File: roles/repo_setup/tasks/get-hash-set-fact-no-collections.yml [hunk 1]
```diff
@@ -8,15 +8,15 @@
         tgh_dlrn_url: "{{ dlrn_baseurl|urlsplit('scheme') }}://{{ dlrn_baseurl|urlsplit('hostname') }}"
     - name: 'get-hash {{ search_tag }} for {{ tgh_release }} on {{ tgh_os_version }} from {{ tgh_dlrn_url }}'
       shell: |
-        repo-setup-get-hash --release {{ tgh_release }} --os-version {{ tgh_os_version }} --tag {{ search_tag }} --dlrn-url {{ tgh_dlrn_url }}
+        repo_setup-get-hash --release {{ tgh_release }} --os-version {{ tgh_os_version }} --tag {{ search_tag }} --dlrn-url {{ tgh_dlrn_url }}
```
Comment (raukadah) [inline roles/repo_setup/tasks/get-hash-set-fact-no-collections.yml]: s/repo_setup-get-hash/repo-setup-get-hash

File: roles/repo_setup/tasks/get-hash-set-fact-no-collections.yml [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
       register: get_hash_result
       until: get_hash_result is success
       retries: 5
   rescue:
     - include_tasks: set_fact_nodepool_rdo_proxy.yml
     - name: 'get-hash {{ search_tag }} for {{ tgh_release }} on {{ tgh_os_version }} from {{ nodepool_rdo_proxy }}'
       shell: |
-        repo-setup-get-hash --release {{ tgh_release }} --os-version {{ tgh_os_version }} --tag {{ search_tag }} --dlrn-url {{ nodepool_rdo_proxy }}
+        repo_setup-get-hash --release {{ tgh_release }} --os-version {{ tgh_os_version }} --tag {{ search_tag }} --dlrn-url {{ nodepool_rdo_proxy }}
```
Comment (raukadah) [inline roles/repo_setup/tasks/get-hash-set-fact-no-collections.yml]: s/repo_setup-get-hash/repo-setup-get-hash

File: roles/repo_setup/tasks/yum-config-generic-no-collections.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@
   shell: >
     export PATH=$PATH:/usr/local/bin && \
     source {{ repo_setup_env_filepath }} && \
-    repo-setup-yum-config repo \
+    repo_setup-yum-config repo \
```
Comment (raukadah) [inline roles/repo_setup/tasks/yum-config-generic-no-collections.yml]: repo-setup-yum-config

**Comment by Sandeepyadav93** [comment]
Looks good to me

**Review by raukadah — APPROVED** [review]

**Pull request merged**